### PR TITLE
Create ubb_bg.py

### LINKED ISSUE #5865: UBB Bank
### Brand name

UBB / ОББ

### Wikidata ID

Q7887555

### Store finder url(s)

https://www.ubb.bg/offices/pins?city_id=&srch_offices=&_=1694684624025

`pin_type` is used to diffrentiate between `offices` and `atms`.

### PR DESCRIPTION
**Pull request by Dimitar5555**
Resolves #5865

I've resorted to some pretty long regexes. Their variable names are not the best, but the comments under the ifs should be enough. Let me know if I can improve them somehow.

### CODE REVIEW

**Comment by Dimitar5555** [comment]
Maybe we need this maioneza cookie?
```js
function sCi(n, value, e, path, domain) {
    var maioneza = n + "=" + escape(value);
    maioneza += ";expires=" + "Wed, 21 Oct 2015 07:28:00 GMT";
    if (path) {
        maioneza += ";path=" + path;
    }
    if (domain) {
        maioneza += ";domain=" + domain;
    }
    maioneza += ';secure';
    maioneza += ';samesite=lax';
    document.cookie = maioneza;
}

function sEx(clf) {
    var today = new Date();
    var expr = new Date(today.getTime() + clf * 24 * 60 * 60 * 1000);
    return expr.toGMTString();
}

function dCi(n) {
    document.cookie = n + '=; Path=/; Expires=-1;samesite=lax';
}

dCi('d41d8cd98f00b204e');
sCi('d41d8cd98f00b204e', '800998ecf8427f', '7200', '/');
```

**Comment by CloCkWeRX** [comment]
Weird.

```
curl 'https://www.ubb.bg/offices/pins' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: en-GB,en;q=0.9,en-US;q=0.8,de;q=0.7,es;q=0.6,fr;q=0.5' \
  -H 'cache-control: max-age=0' \
  -H 'cookie: vxcaccess=2e8960ee73dce2bbd007cee31a54a9e4' \
  -H 'dnt: 1' \
  -H 'priority: u=0, i' \
  -H 'referer: https://www.ubb.bg/offices/pins' \
  -H 'sec-ch-ua: "Not)A;Brand";v="99", "Google Chrome";v="127", "Chromium";v="127"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: same-origin' \
  -H 'sec-fetch-user: ?1' \
  -H 'sec-gpc: 1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36'
```
Yields:
```
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36'
<html lang="en-US" charset="UTF-8"> <head><meta charset="UTF-8"><meta http-equiv="refresh" content="1" /><meta http-equiv="cache-control" content="max-age=0" /><meta http-equiv="cache-control" content="no-cache" /><meta http-equiv="expires" content="-1" /><meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT" /><meta http-equiv="pragma" content="no-cache" /></head> <script type="text/javascript"> function sCi(n, value, e, path, domain){ var maioneza = n + "=" + escape(value); maioneza += ";expires=" + "Wed, 21 Oct 2015 07:28:00 GMT"; if(path){ maioneza += ";path=" + path; } if(domain){ maioneza += ";domain=" + domain; } maioneza += ';secure'; maioneza += ';samesite=lax'; document.cookie = maioneza; } function sEx(clf){ var today = new Date(); var expr = new Date(today.getTime() + clf * 24 * 60 * 60 * 1000); return  expr.toGMTString(); } function dCi(n){ document.cookie = n +'=; Path=/; Expires=-1;samesite=lax'; } </script> <script type="text/javascript"> dCi('d41d8cd98f00b204e'); sCi('d41d8cd98f00b204e', '800998ecf8427f', '7200', '/'); </script> <body>.</body> </html>
```

File: locations/spiders/ubb_bg.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+class UbbBGSpider(Spider):
+    name = "ubb_bg"
+    item_attributes = {"brand": "Обединена българска банка", "brand_wikidata": "Q7887555"}
+    start_urls = ["https://www.ubb.bg/offices/pins"]
+    user_agent = FIREFOX_LATEST
+    custom_settings = {
+        "ROBOTSTXT_OBEY": False,
+        "DEFAULT_REQUEST_HEADERS": {"Accept": "application/json, text/javascript, */*"},
+    }
```
Comment (CloCkWeRX) [inline locations/spiders/ubb_bg.py:21]:
```suggestion
    }
    requires_proxy = True
```

**Comment by CloCkWeRX** [comment]
I think add     requires_proxy = True, and paste your results from running locally - may be suitable to merge at that point.

File: locations/spiders/ubb_bg.py (hunk 1 above)
Comment (Dimitar5555) [reply locations/spiders/ubb_bg.py:20]: Thank you! It works as expected now. I assume it also needed the cookie, which is why it didn't work with the proxy the first time.

**Comment by Dimitar5555** [comment]
> and paste your results from running locally

https://gist.github.com/Dimitar5555/adb289132c39b42b859a312f1e88df01

File: locations/spiders/ubb_bg.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
```
Comment (Cj-Malone) [inline locations/spiders/ubb_bg.py:20]: Do we need to ignore robots.txt? It doesn't look like we're blocked.

https://www.ubb.bg/robots.txt

Also headers are probably not needed.

```suggestion
```

**Pull request merged**